Add bubble chart legend, links, tooltip

aabdaab opened the pull request:
This PR wraps up the sections bubble chart. In more detail:

* Adds Bubble Chart legend
![image](https://user-images.githubusercontent.com/6136899/68691556-db78fb80-056b-11ea-9383-331c571ff7f2.png)
* Adds links to sections
* Fixes typography, small style issues
* Add click handlers on single bubbles - link to companies pages
* Add `BaseTooltip` component
* Add tooltip for Market Cap
* Remove the last column `4STAR` - after talking to Fausto I learned that 4STAR companies should be included in `Level 4`

<img width="1424" alt="Screenshot 2019-11-12 at 16 50 36" src="https://user-images.githubusercontent.com/6136899/68691998-9bfedf00-056c-11ea-8043-34580428d00a.png">

tsubik commented:
Nice! I'm on it.

tsubik commented on app/models/company.rb:
```diff
@@ -68,4 +68,8 @@ def isin_array
   end
+
+  def is_4_star?
+    latest_mq_assessment.level.eql?('4STAR')
```
You could use `level.eql('4STAR')`

tsubik commented on app/services/api/charts/sector.rb:
```diff
@@ -29,6 +29,7 @@ def companies_summaries_by_level
 
       def companies_market_cap_by_sector
```
we gonna need some test coverage for this method at some point

tsubik commented:
We have to take a look at those methods which are aggregating the data as I see tons of SELECT queries in the log when entering sectors' index.

aabdaab commented:
About the hovering issue, I can't reproduce it :/ Maybe it's connected to some scaling? were your dev tools open on the right side of the screen?
![screencast 2019-11-12 17-23-21](https://user-images.githubusercontent.com/6136899/68694608-63153900-0571-11ea-8089-9781ccdee1ee.gif)

aabdaab commented:
I also noticed, there's a centering issue, I'll take a look at it

tsubik commented:
I still have it, on Chrome and Firefox, no dev tools opened. I see that svg in bubble chart cell is bigger than cell itself. That could be the issue right, it's overlapping another cell.

rapzo commented on app/assets/stylesheets/settings.scss:
```diff
@@ -1,3 +1,9 @@
 
-$font-size-36: 2.25rem;
\ No newline at end of file
+$font-size-36: 2.25rem; // 1rem = 16px
```
This file can be deleted since it is kind of redundant.
`$font-size-36` should replace, probably, one of the default sizes, maybe $size-3.

rapzo commented on app/assets/stylesheets/tpi/_bubble-chart.scss:
```diff
@@ -55,4 +56,92 @@ $cell-height: 55px;
+
+.bubble-chart__title {
+  color: $grey-dark;
+  font-size: 16px;
```
`$size-6`

rapzo commented on app/assets/stylesheets/tpi/_base-tooltip.scss:
```diff
@@ -0,0 +1,19 @@
+  min-width: 200px;
+  min-height: 110px;
+  background-color: $dark;
+  color: rgba(255,255,255,0.9);
```
`rgba($white, 0.9)` works and that way colors don't get loose around.

rapzo commented on app/javascript/components/BaseTooltip.js:
```diff
@@ -0,0 +1,22 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
```
Missing props validation?

tsubik commented:
Needs rebasing

tsubik commented:
Can we easy remove that effect of initial centering bubbles? It's slowing down my browser :P

aabdaab commented:
I don't think so, I think that's how `d3-force` works 🤔 trying to "attract" the bubbles to the y axis

aabdaab commented:
see here: https://bl.ocks.org/d3indepth/fee5ce57c3fc3e94c3332577d1415df4 - the `cy` and `cx` are recalculating for some time to align the bubbles

tsubik commented:
Ok, there is no time for it now. But maybe there is a room for later improvement

tsubik commented:
It's kinda slow, not sure if that's React or because so many separate cells.

tsubik commented:
I will try to reboot my machine and open it once again :D

aabdaab commented:
* I addressed the CR comments
* Fixed the hovering and centering issue with of the bubbles

I'm looking into performance improvements to fix the N+1 queries, but maybe you think that could be in a different PR?

tsubik commented:
Personally, I would do it here, if you think it won't take that long, but if we need to do bigger refactor then separate one.

simaob commented:
think this is ready for review again.

tsubik commented:
I'm on it

tsubik commented:
We need materialized views to fix performance issues, but that is for another story.
